generators/entity/templates/src/test/java/package/web/rest/_EntityResourceIntTest.java

```diff
@@ -361,6 +382,35 @@ public void initTest() {<% if (databaseType == 'mongodb' || databaseType == 'cas
 
         // Validate the database is empty
         List<<%= entityClass %>> <%= entityInstancePlural %> = <%= entityInstance %>Repository.findAll();
-        assertThat(<%= entityInstancePlural %>).hasSize(databaseSizeBeforeDelete - 1);
-    }
+        assertThat(<%= entityInstancePlural %>).hasSize(databaseSizeBeforeDelete - 1);<% if (searchEngine == 'elasticsearch') { %>
+
+        // Validate tElasticSearch is empty
+        boolean <%= entityInstance %>ExistsInEs = <%= entityInstance %>SearchRepository.exists(<%= entityInstance %>.getId());
```

Comment:
Inversing elasticSearch validation and database validation works.
Replcaing `List<<%= entityClass %>> <%= entityInstancePlural %>` by `List<<%= entityClass %>> <%= entityInstancePlural %>Database` works too.